server/src/modules/clubs/application/use-cases/get-all-clubs/get-all-clubs-controller.ts

```diff
@@ -0,0 +1,21 @@
+import express from 'express'
+import { BaseController } from '../../../../../shared/app/base-controller'
+import { ClubRepo } from '../../../infra/repos/club-repo'
+import { ClubDTO } from '../../../mappers/club-dto'
+import { ClubMap } from '../../../mappers/club-map'
+
+export class GetAllClubsController extends BaseController {
+  constructor(private clubRepo: ClubRepo) {
+    super()
+  }
+
+  async execute(_req: express.Request, res: express.Response): Promise<express.Response> {
+    try {
+      const clubs = await this.clubRepo.getAllClubs()
+      const clubDTOs: Array<ClubDTO> = clubs.map((club) => ClubMap.toDTO(club))
+      return this.ok(res, clubDTOs)
+    } catch (err) {
```

Comment:
`try/catch` is what we're trying to avoid with the `Result` type. we want to encode fallibility into our type system. But right now our repos are returning promises. I also see that the initial code I put in uses a try catch too (`create-user-use-case.ts` when calling the repo.

We need to have a team discussion on how to handle fallibility from our repo layer.

1. Continue with try/catch?
2. Return `Result<T>` and modify our `Results` to have async/await support . Perhaps this is when we install a production-ready result-type package.

cc: @xujustinj

Reply:
We wouldn't even need to wrap `.executeImpl` in a try/catch if our repos return Results instead of promises that throw. We would then need results that are `.then`able/`await`able.

Reply:
if we don't wrap it, wouldn't an unexpected error crash the entire server?

Reply:
Final Decision:

-Remove try/catch blocks from use cases and controllers
-All repo methods contain their own try/catch blocks, and only Results are returned